src/test/java/org/opensearch/sdk/rest/TestBaseExtensionRestHandler.java

```diff
@@ -0,0 +1,267 @@
+        assertEquals(RestStatus.OK, response.status());
+        assertEquals("{\"success\":\"bar\"}", response.content().utf8ToString());
+    }
+
+    @Test
+    public void testJsonDeprecatedResponse() {
+        RestRequest successfulRequest = TestSDKRestRequest.createTestRestRequest(
+            Method.GET,
+            "/deprecated/foo",
+            "/deprecated/foo",
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            null,
+            new BytesArray("bar".getBytes(StandardCharsets.UTF_8)),
+            "",
+            null
+        );
+        ExtensionRestResponse response = handler.handleRequest(successfulRequest);
+        assertEquals(RestStatus.OK, response.status());
+        assertEquals("{\"success\":\"bar\"}", response.content().utf8ToString());
+    }
+
+    @Test
+    public void testJsonReplacedResponseGet() {
+        RestRequest successfulRequestOld = TestSDKRestRequest.createTestRestRequest(
+            Method.GET,
+            "/old/foo",
+            "/old/foo",
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            null,
+            new BytesArray("bar".getBytes(StandardCharsets.UTF_8)),
+            "",
+            null
+        );
+        RestRequest successfulRequestNew = TestSDKRestRequest.createTestRestRequest(
+            Method.GET,
+            "/new/foo",
+            "/new/foo",
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            null,
+            new BytesArray("bar".getBytes(StandardCharsets.UTF_8)),
+            "",
+            null
+        );
+        ExtensionRestResponse response = handler.handleRequest(successfulRequestOld);
+        assertEquals(RestStatus.OK, response.status());
+        assertEquals("{\"success\":\"bar\"}", response.content().utf8ToString());
+
+        response = handler.handleRequest(successfulRequestNew);
+        assertEquals(RestStatus.OK, response.status());
+        assertEquals("{\"success\":\"bar\"}", response.content().utf8ToString());
+    }
+
+    @Test
+    public void testJsonReplacedResponsePut() {
+        RestRequest successfulRequestOld = TestSDKRestRequest.createTestRestRequest(
+            Method.PUT,
+            "/old/put/foo",
+            "/old/put/foo",
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            null,
+            new BytesArray("bar".getBytes(StandardCharsets.UTF_8)),
+            "",
+            null
+        );
+        RestRequest successfulRequestNew = TestSDKRestRequest.createTestRestRequest(
+            Method.PUT,
+            "/new/put/foo",
+            "/new/put/foo",
+            Collections.emptyMap(),
+            Collections.emptyMap(),
+            null,
+            new BytesArray("bar".getBytes(StandardCharsets.UTF_8)),
+            "",
+            null
+        );
+        ExtensionRestResponse response = handler.handleRequest(successfulRequestOld);
```

Comment:
Can we test here whether the old request has been replaced to the new one? Mostly by checking the URI path?

Reply:
There is no actual "replacement" happening.  A replaced route is just syntactic sugar for simultaneously registering a deprecated route and a route both pointing to the same handler, which is exactly what's been tested.  We could add another test of the deprecation message on the old handler.

Reply:
Added new tests in the registry that tested these routes were added to both deprecated and normal lists and tested the deprecation message too. :)